src/safeposix/syscalls/fs_constants.rs

```diff
@@ -139,6 +141,28 @@ pub struct StatData {
   pub st_ctim: (u64, u64)
 }
 
+//derive eq attributes for testing whether the structs equal other fsdata structs from stat/fstat
+#[derive(Eq, PartialEq, Default)]
+#[repr(C)]
+pub struct FSData {
+  pub f_type: u32,
+  pub f_bsize: u32,
+  pub f_blocks: u32,
+  pub f_bfree: u64,
+  pub f_bavail: u64,
+  //total files in the file system -- should be infinite
+  pub f_files: u64,
+  //free files in the file system -- should be infinite
+  pub f_ffiles: u64,
+  pub f_fsid: u64,
+  //not really a limit for naming, but 254 works
+  pub f_namelen: u32,
+  //arbitrary val for blocksize as well
+  pub f_frsize: u32,
```

Comment:
namelen and frsize should be u64 as well, unless tests on your system yield different results (in which case we'll have to figure something out).